src/hardware/adlib_gold.cpp

```diff
@@ -0,0 +1,352 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2022  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#include "adlib_gold.h"
+#include "checks.h"
+
+CHECK_NARROWING();
+
+// Yamaha YM7128B Surround Processor emulation
+// -------------------------------------------
+
+SurroundProcessor::SurroundProcessor(const uint16_t sample_rate) : chip(nullptr)
+{
+	chip = (YM7128B_ChipIdeal *)malloc(sizeof(YM7128B_ChipIdeal));
```

Comment:
Would be nice to make `YM7128B_ChipIdeal` just a normal member object of the `SurroundProcessor` class, because it's a tiny little struct and avoid this low-level malloc'ing of raw bytes.

Reply:
You're right. I just copied this from the example code and never thought about it again.